java/ql/lib/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -182,6 +204,695 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ".map;AbstractHashedMap;true;createEntrySetIterator;;;MapValue of Element of ReturnValue; MapValue of Argument[-1];value",
+          ".map;AbstractHashedMap;true;createKeySetIterator;;;Element of ReturnValue; MapKey of Argument[-1];value",
+          ".map;AbstractHashedMap;true;createValuesIterator;;;Element of ReturnValue; MapValue of Argument[-1];value",
+          ".map;AbstractHashedMap$HashEntry;true;HashEntry;;;Argument[2];MapKey of Argument[-1];value",
+          ".map;AbstractHashedMap$HashEntry;true;HashEntry;;;Argument[3];MapValue of Argument[-1];value",
+          ".map;AbstractLinkedMap;true;AbstractLinkedMap;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;AbstractLinkedMap;true;AbstractLinkedMap;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;AbstractMapDecorator;true;AbstractMapDecorator;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;AbstractMapDecorator;true;AbstractMapDecorator;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;AbstractMapDecorator;true;decorated;;;MapKey of Argument[-1];MapKey of Argument[0];value",
+          ".map;AbstractMapDecorator;true;decorated;;;MapValue of Argument[-1];MapValue of Argument[0];value",
+          ".map;AbstractOrderedMapDecorator;true;AbstractOrderedMapDecorator;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;AbstractOrderedMapDecorator;true;AbstractOrderedMapDecorator;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;AbstractSortedMapDecorator;true;AbstractSortedMapDecorator;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;AbstractSortedMapDecorator;true;AbstractSortedMapDecorator;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;AbstractSortedMapDecorator$SortedMapIterator;true;SortedMapIterator;;;MapKey of Element of Argument[0];Element of Argument[-1];value",
+          ".map;AbstractSortedMapDecorator$SortedMapIterator;true;SortedMapIterator;;;MapValue of Element of Argument[0];MapValue of Argument[-1];value",
+          ".map;CaseInsensitiveMap;true;CaseInsensitiveMap;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;CaseInsensitiveMap;true;CaseInsensitiveMap;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;CompositeMap;true;CompositeMap;(Map,Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;CompositeMap;true;CompositeMap;(Map,Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;CompositeMap;true;CompositeMap;(Map,Map);;MapKey of Argument[1];MapKey of Argument[-1];value",
+          ".map;CompositeMap;true;CompositeMap;(Map,Map);;MapValue of Argument[1];MapValue of Argument[-1];value",
+          ".map;CompositeMap;true;CompositeMap;(Map,Map,MapMutator);;MapKey of Argument[0];MapKey of Argument[-1];value",
```

Comment:
I wonder if the variations including the `MapMutator` should be handled as `taint`

Reply:
My impression was that the `MapMutator` was intended to be used to decide which of the array of maps to do the mutation operation on, which we don't care about in our model.